js translation -- performance improvment

graceguo-supercat opened the pull request.

lxzhangchao committed:
Chinese page

lxzhangchao committed:
Using react-intl-universal to improve multi language in react page

lxzhangchao committed:
Using react-intl-universal to improve multi language in react page

lxzhangchao committed:
Merge pull request #1 from lxzhangchao/ch_react_page

This is use react-intl-universal to improve multi language in react page

lxzhangchao committed:
react_intl_universal

lxzhangchao committed:
Merge pull request #2 from lxzhangchao/ch_react_page

Using react-intl-universal to improve multi language in react page

lxzhangchao committed:
change

lxzhangchao committed:
Merge pull request #3 from lxzhangchao/ch_react_page

use react-intl-universal to improve multi language in react page

lxzhangchao committed:
change

lxzhangchao committed:
change

lxzhangchao committed:
change

lxzhangchao committed:
change

lxzhangchao committed:
change

lxzhangchao committed:
change

lxzhangchao committed:
Merge pull request #4 from lxzhangchao/ch_react_page

Ch react page

lxzhangchao committed:
merge

lxzhangchao committed:
multiple page in js

lxzhangchao committed:
Conflict

lxzhangchao committed:
merge

lxzhangchao committed:
merge

lxzhangchao committed:
merge

lxzhangchao committed:
merge

lxzhangchao committed:
Merge branch 'master' of https://github.com/apache/incubator-superset into mu_js_page

lxzhangchao committed:
Js Translations

lxzhangchao committed:
JS Translation

lxzhangchao committed:
JS Translations

lxzhangchao committed:
Js translation

lxzhangchao committed:
JS translations

lxzhangchao committed:
JS translations

lxzhangchao committed:
Js translaion

lxzhangchao committed:
JS en Translation

lxzhangchao committed:
JS Translation

lxzhangchao committed:
Merge branch 'master' of https://github.com/apache/incubator-superset into mu_js_page

lxzhangchao committed:
upgrade document

Fixing the damn build (#3179)

* Fixing the build

* Going deeper

[bugfix] only filterable columns should show up in FilterBox list (#3105)

* [bugfix] only filterable columns should show up in FilterBox list

* Touchups

Datasource cannot be empty (#3035)

add title description to model view (#3045)

* add title description to model view

* add missing import

Add 'show/hide totals' option to pivot table vis (#3101)

[bugfix] numeric value for date fields in table viz (#3036)

Bug was present only when using the NOT GROUPED BY option

fixes https://github.com/ApacheInfra/superset/issues/3027

fix hive.fetch_logs (#2968)

add Zalando to the list of organizations (#3171)

docs: fixup installation examples code indentation (#3169)

[bugfix] fix bar order (#3180)

[bugfix] visualize flow error: 'Metric x is not valid' (#3181)

The metric name in the frontend doesn't match the one generated on the
backend. It turns out the explore view will default to the first
metric so specifying one isn't needed.

Fix the segment interval for pulling metadata (#3174)

The end of the interval would be on the truncated today date, which
means that you will exclude today. If your realtime ingestion job
runs shorter than a day, the metadata cannot be pulled from the
druid cluster.

Bump cryptography to 1.9 (#3065)

As 1.7.2 doesn't compile here with openssl 1.1.0f

Escaping the user's SQL in the explore view (#3186)

* Escaping the user's SQL in the explore view

When executing SQL from SQL Lab, we use a lower level API to the
database which doesn't require escaping the SQL. When going through
the explore view, the stack chain leading to the same method may need
escaping depending on how the DBAPI driver is written, and that is the
case for Presto (and perhaps other drivers).

* Using regex to avoid doubling doubles

[sqllab] improve Hive support (#3187)

* [sqllab] improve Hive support

* Fix "Transport not open" bug
* Getting progress bar to show
* Bump pyhive to 0.4.0
* Getting [Track Job] button to show

* Fix testzz

Add BigQuery engine specifications (#3193)

As contributed by @mxmzdlv on issue #945

[bugfix] fix merge conflict that broke Hive support (#3196)

Adding 'apache' to docs (#3194)

[druid] Allow custom druid postaggregators (#3146)

* [druid] Allow custom druid postaggregators

Also, fix the postaggregation for approxHistogram quantiles so it adds
the dependent field and that can show up in the graphs/tables.

In general, postAggregators add significant power, we should probably
support including custom postAggregators. Plywood has standard
postAggregators here, and a customAggregator escape hatch that allows
you to define custom postAggregators.

This commit adds a similar capability for Superset and a additional
field/fields/fieldName breakdown of the typical naming for dependent
aggregations, which should make it significantly easier to develop
approxHistogram and custom postAggregation-required dashboards.

* [druid] Minor style cleanup in tests file.

* [druid] Apply code review suggestions

* break out CustomPostAggregator into separate class. This just cleans
  up the creation of the postaggregator a little bit.
* minor style issues.
* move the function around so the git diff is more readable

add combine config for metrics in pivot table (#3086)

* add combine config for metrics in pivot table

* change method to stack/unstack

* update backendSync

Autofocus search input in VizTypeControl modal onEnter (#2929)

Speed up JS build time (#3203)

Also bumping a few related libs

JS Translation

JS translations

js translation

fix issue 3204 (#3205)

[bugfix] capture Hive job_id pre-url transformation (#3213)

js translation

fix issue 3204 (#3205)

[bugfix] capture Hive job_id pre-url transformation (#3213)

[docs] update url in CONTRIBUTING.md (#3212)

[sqllab/cosmetics] add margin-top for labels in query history (#3222)

[explore] nvd3 sort values in rich tooltip (#3197)

[sqllab] fix UI shows 'The query returned no results' momentarily (#3214)

this is visible when running async queries between the fetching and
success state as the rows are getting cached in the component

[explore] DatasourceControl to pick datasource in modal (#3210)

* [explore] DatasourceControl to pick datasource in modal

Makes it easier to change datasource, also makes it such that the list
of all datasources doesn't need to be loaded upfront.

* Adding more metadata

lxzhangchao committed:
mergeCode

lxzhangchao committed:
Js translation

lxzhangchao committed:
Merge branch 'master' of https://github.com/apache/incubator-superset into js_po_translation

lxzhangchao committed:
Js tran

lxzhangchao committed:
js tran

lxzhangchao committed:
Merge branch 'master' of https://github.com/apache/incubator-superset into js_po_translation

lxzhangchao committed:
Merge branch 'master' of https://github.com/apache/incubator-superset into js_po_translation

lxzhangchao committed:
js trans

lxzhangchao committed:
Merge branch 'master' of https://github.com/apache/incubator-superset into js_po_translation

lxzhangchao committed:
js trans

lxzhangchao committed:
js tran

lxzhangchao committed:
merge

lxzhangchao committed:
js trans

lxzhangchao committed:
js trans

lxzhangchao committed:
js tran

lxzhangchao committed:
merge

lxzhangchao committed:
js translation

lxzhangchao committed:
js trans

lxzhangchao committed:
merge

lxzhangchao committed:
js translation

lxzhangchao committed:
Merge branch 'master' of https://github.com/apache/incubator-superset into js_po_translation

lxzhangchao committed:
merge Conflict

Commit:
try load language pack async

mistercrunch committed:
Backend translations things

Commit:
Merge remote-tracking branch 'upstream/master' into i18n

Commit:
create language pack inside common data

Commit:
Merge remote-tracking branch 'upstream/master' into i18n

Commit:
Merge remote-tracking branch 'upstream/master' into i18n

Commit:
performance improvement for js i18n.

- js bundle should not contain localized content
- we populate translation content from server-side, in boostrap.common.language_pack
- in client-side, use promise to wrap around translation content. text will be translated after translation content arrived/parsed.
- fix linting

xrmx commented on superset/assets/javascripts/explore/stores/visTypes.js:
```diff
@@ -21,31 +25,37 @@ export const sections = {
   },
   NVD3TimeSeries: [
     {
       label: 'Query',
```
Other labels have parens now,out of curiosity why?

xrmx replied:
Also why we are duplicating label -> labelDisplay amd description -> descriptionDisplay?

xrmx commented on superset/translations/utils.py:
```diff
@@ -0,0 +1,33 @@
+    pack = ALL_LANGUAGE_PACKS.get(locale)
+    if not pack:
+        filename = DIR + '/{}/LC_MESSAGES/messages.json'.format(locale)
+        with open(filename) as f:
```
this will raise an exception if the file does not exist

mistercrunch commented on superset/views/core.py:
```diff
@@ -2293,6 +2294,13 @@ def sqllab(self):
         )
+
+    @has_access
+    @expose("/rest/api/get_locale", methods=['GET', 'POST'])
```
unused endpoint, remove

mistercrunch commented on superset/assets/package.json:
```diff
@@ -120,6 +124,7 @@
     "less": "^2.6.1",
     "less-loader": "^4.0.3",
     "mocha": "^3.2.0",
+    "po-catalog-loader": "^1.2.0",
```
unused?

graceguo-supercat replied:
removed.

mistercrunch commented on superset/assets/package.json:
```diff
@@ -111,6 +114,7 @@
     "exports-loader": "^0.6.3",
     "extract-text-webpack-plugin": "3.0.0",
     "file-loader": "^0.11.1",
+    "gettext-parser": "^1.3.0",
```
unused?

graceguo-supercat replied:
removed

mistercrunch commented on superset/assets/javascripts/locales.jsx:
```diff
@@ -0,0 +1,176 @@
+let i18n = null;
+let i18nPromise = null;
+
+function formatForReact(formatString, args) {
```
I don't understand what this function does it appears to me that `args` can be React elements, but I don't understand the use case.

mistercrunch commented on superset/assets/javascripts/explore/stores/visTypes.js:
```diff
@@ -1,18 +1,22 @@
   druidTimeSeries: {
-    label: 'Time',
+    label: ('Time'),
+    labelDisplay: t('Time'),
```
I hate the fact that `labelDisplay` and `descriptionDisplay` have a negative impact on our app's memory footprint despite the fact that it's only used for flask-babel to find the strings to translate. What about using a noop `t` function in this context?

From what I understand, you could simply go `const t = () => {};` instead of `import { t } from '../../locales';` and babel should still pick up all the calls. Then you can get rid of all the `*Display` entries. Wouldn't that work? If it works, we should do that and make sure to add a comment that explains the problematic of dealing with i18n strings in module scope. The comment should read something like:

```
For i18n strings that are defined in module scope, we cannot have the guarantee that the translation dictionary hasn't been initialized at module-import-time. In those cases we use a noop `t()` function so that the strings can get detected by Python's babel, but those strings need to be translated at runtime where they are used as variables.
```

graceguo-supercat replied:
removed all labelDisplay and descriptionDisplay.
created a new module, i18n, to parse and create Jed object. since i18n is dependency of locales.jsx, these modules will be loaded before vizType, control.jsx etc.

Commit:
fix Timer unit test

Commit:
1. add global hook for all tests, to make translation pack avaialble before each test starts.
2. fix unit test for Timer component
3. remove noused method get_locale, and modules
4. fix page reload after user change page language

Commit:
parse and build i18n dictionary as a module

zhangwenwu commented:
When will this function be available?

Commit:
Merge remote-tracking branch 'upstream/master' into i18n

Commit:
Merge remote-tracking branch 'upstream/master' into i18n

Commit:
fix sync-backend task, which should run without DOM

graceguo-supercat commented on superset/views/core.py:
```diff
@@ -2293,6 +2294,13 @@ def sqllab(self):
+
+    @has_access
+    @expose("/rest/api/get_locale", methods=['GET', 'POST'])
+    def get_locale(self):
```
removed

graceguo-supercat commented on superset/assets/webpack.config.js:
```diff
@@ -116,6 +116,14 @@ const config = {
         loader: 'transform/cacheable?brfs',
       },
+      {
+        test: /\.po$/,
```
removed

graceguo-supercat commented on superset/assets/spec/javascripts/sqllab/VisualizeModal_spec.jsx:
```diff
@@ -22,6 +22,8 @@ global.notify = {
   error: () => {},
 };
 
+require('../helper');
```
removed.

zhangwenwu commented:
When will this function be released?

xrmx commented:
@zhangwenwu when it's ready

zhangwenwu commented:
good，thank you

zwcwu5 commented:
how long will be it done?

Commit:
Merge remote-tracking branch 'upstream/master' into i18n
